### DISCUSSION TITLE
Like expression with a case statement

### DISCUSSION THREAD

**Discussion by davidherdu** [doctrine/doctrine2#9177] [Support Questions]
Hi, I'm building the following query with a query builder
```
SELECT u, CASE WHEN u.roles LIKE '%ROLE_SUPER_ADMIN%' THEN 'Admin' ELSE '' END AS role 
FROM App\Entity\User u 
WHERE (CASE WHEN u.roles LIKE '%ROLE_SUPER_ADMIN%' THEN 'Admin' ELSE '' END LIKE :filter0_0)
```
and I got the error `[Syntax Error] line 0, col 936: Error: Expected =, <, <=, <>, >, >=, !=, got 'LIKE'`
That query works perfectly in sequel pro, if I replace the last `LIKE` by `=` it works in doctrine as well.

**Comment by cbichis** [comment]
I confirm the issue.
